graphrag/query/input/retrieval/relationships.py

```diff
@@ -7,22 +7,41 @@
 
 import pandas as pd
 
+from common.graph_db_client import GraphDBClient
 from graphrag.model import Entity, Relationship
 
+from graphrag.query.input.loaders.dfs import read_relationships
 
 def get_in_network_relationships(
     selected_entities: list[Entity],
     relationships: list[Relationship],
     ranking_attribute: str = "rank",
+    graphdb_client: GraphDBClient|None=None,
 ) -> list[Relationship]:
     """Get all directed relationships between selected entities, sorted by ranking_attribute."""
     selected_entity_names = [entity.title for entity in selected_entities]
-    selected_relationships = [
-        relationship
-        for relationship in relationships
-        if relationship.source in selected_entity_names
-        and relationship.target in selected_entity_names
-    ]
+    if not graphdb_client:
+        selected_relationships = [
+            relationship
+            for relationship in relationships
+            if relationship.source in selected_entity_names
+            and relationship.target in selected_entity_names
+        ]
+    else:
+        selected_relationships_result = graphdb_client._client.submit(
```

Comment:
I like this, but I think the query text should be wrapped in a function that takes in selected_entity_names amd retirms the result